ospfd/ospf_vty.c

```diff
@@ -10648,28 +10659,42 @@ static void ospf_interface_clear(struct interface *ifp)
 
 DEFUN (clear_ip_ospf_interface,
        clear_ip_ospf_interface_cmd,
-       "clear ip ospf interface [IFNAME]",
+       "clear ip ospf [vrf <NAME>] interface [IFNAME]",
        CLEAR_STR
        IP_STR
        "OSPF information\n"
+       VRF_CMD_HELP_STR
        "Interface information\n"
        "Interface name\n")
 {
-	int idx_ifname = 4;
+	int idx_ifname = 6;
+	int idx_vrf = 0;
 	struct interface *ifp;
 	struct listnode *node;
 	struct ospf *ospf = NULL;
+	char *vrf_name = NULL;
+	vrf_id_t vrf_id = VRF_DEFAULT;
+	struct vrf *vrf = NULL;
 
-	if (argc == 4) /* Clear all the ospfv2 interfaces. */
-	{
+	if (argv_find(argv, argc, "vrf", &idx_vrf))
+		vrf_name = argv[idx_vrf + 1]->arg;
+	if (vrf_name == NULL || strmatch(vrf_name, VRF_DEFAULT_NAME))
+		vrf_name = NULL;
+	if (vrf_name)
+		vrf = vrf_lookup_by_name(vrf_name);
+	if (vrf)
+		vrf_id = vrf->vrf_id;
+
+	if (!argv_find(argv, argc, "IFNAME", &idx_ifname)) {
+		/* Clear all the ospfv2 interfaces. */
```

Comment:
If a VRF is specified, shouldn't we clear the OSPF interfaces from that VRF only?